framework/src/base/FEProblemBase.C

```diff
@@ -1439,6 +1440,26 @@ FEProblemBase::getFunction(const std::string & name, THREAD_ID tid)
 }
 
 void
+FEProblemBase::addDistribution(std::string type, const std::string & name, InputParameters parameters)
+{
+  setInputParametersFEProblem(parameters);
+  for (THREAD_ID tid = 0; tid < libMesh::n_threads(); tid++)
+  {
+    std::shared_ptr<Distribution> dist = _factory.create<Distribution>(type, name, parameters, tid);
+    _distributions.addObject(dist,tid);
+  }
+}
+
+Distribution &
+FEProblemBase::getDistribution(const std::string & name, THREAD_ID tid)
```

Comment:
Is it possible for this to return a `const` reference to the distribution? It doesn't seem that calling methods (e.g., pdf, cdf, etc.) should change the state of the object, is that true. If so, then this should be const and the methods that will be accessed by users be set to const methods.